util/dvsim/dvsim.py

```diff
@@ -298,32 +318,27 @@ def main():
 
     parser.add_argument("-c",
                         "--cov",
-                        default=False,
                         action='store_true',
                         help="turn on coverage collection")
 
     parser.add_argument(
         "--cov-merge-previous",
-        default=False,
         action='store_true',
         help="""Applicable when --cov switch is enabled. If a previous cov
                 database directory exists, this switch will cause it to be merged with
                 the current cov database.""")
 
     parser.add_argument(
         "--cov-analyze",
-        default=False,
         action='store_true',
         help="Analyze the coverage from the last regression result.")
 
     parser.add_argument("-p",
                         "--profile",
-                        default="none",
-                        metavar="time|mem",
+                        choices=["time", "mem"],
```

Comment:
The original code supported the following behavior:
- If `--profile` switch not passed, then dont profile the sim
- If `--profile` switch passed, profile the sim for time by default
- Allow `--profile` to accept one optional arg to profile it differently (VCS supports profiling by memory consumption)

Reply:
I don't think you've correctly described the code that used to exist (you'd need `nargs='?'` for that behaviour), but I can fix the patch to behave as you say.

Reply:
This is working as intended now.